modules/bastion/main.tf

```
@@ -0,0 +1,145 @@
+locals {
+  enabled         = module.this.enabled
+  route53_enabled = local.enabled && var.associate_public_ip_address && var.custom_bastion_hostname != null && var.vanity_domain != null
+
+  vpc_id                 = module.vpc.outputs.vpc_id
+  vpc_private_subnet_ids = module.vpc.outputs.private_subnet_ids
+  vpc_public_subnet_ids  = module.vpc.outputs.public_subnet_ids
+  vpc_subnet_ids         = var.associate_public_ip_address ? local.vpc_public_subnet_ids : local.vpc_private_subnet_ids
+
+  bastion_subnet = slice(local.vpc_subnet_ids, 0, 1)
+  bastion_az     = module.vpc.outputs.availability_zones[0]
+
+  userdata_template             = "${path.module}/templates/user-data.sh"
+  container_template            = "${path.module}/templates/container.sh"
+  container_cloud_init_template = "${path.module}/templates/container-cloud-init.sh"
+
+  cloudwatch_agent_config = templatefile(local.container_cloud_init_template, {
+    content = jsonencode({
+      write_files = [
+        {
+          path        = "/tmp/container.sh"
+          permissions = "0755"
+          owner       = "root:root"
+          encoding    = "b64"
+          content = base64encode(templatefile(local.container_template, {
+            region            = var.region
+            image_repository  = var.image_repository
+            image_container   = var.image_container
+            container_command = var.container_command
+          }))
+        },
+      ]
+    })
+  })
+}
+
+data "aws_route53_zone" "vanity" {
+  count = local.route53_enabled ? 1 : 0
+  name  = var.vanity_domain
+}
+
+resource "aws_route53_record" "default" {
+  count   = local.route53_enabled ? 1 : 0
+  zone_id = join("", data.aws_route53_zone.vanity.*.zone_id)
+  name    = "${var.custom_bastion_hostname}.${var.vanity_domain}"
+  type    = "A"
+  ttl     = "300"
+  records = aws_eip.static.*.public_ip
+}
+
+module "aws_key_pair" {
+  source              = "cloudposse/key-pair/aws"
+  version             = "0.18.0"
```

Please bump `"cloudposse/key-pair/aws"`

```suggestion
  version             = "0.18.1"
```
https://github.com/cloudposse/terraform-aws-key-pair/pull/65